trillian_log_api.proto

```diff
@@ -41,6 +41,14 @@ service TrillianLog {
       };
     }
 
+    // Adds a single leaf with an assigned sequence number.
+    rpc AddSequencedLeaf (AddSequencedLeafRequest) returns (AddSequencedLeafResponse) {
+      option (google.api.http) = {
+        post: "/v1beta1/logs/{log_id}/leaves:sequenced"
```

Comment:
I could make it `"/v1beta1/logs/{log_id}/leaves/{leaf_index}"` instead (as @gdbelvin suggested). This would require an extra `leaf_index` field in `AddSequencedLeafRequest`, and an "Output only" semantics on `LogLeaf.LeafIndex`. To be consistent, the `AddSequencedLeavesRequest` for the batch API would also need to ignore `LogLeaf.LeafIndex` and keep the indices aside as follows:

```
message AddSequencedLeavesRequest {
    int64 log_id = 1;
    repeated int64 leaf_indices  = 2;
    repeated LogLeaf leaves = 3;
}
```
or
```
message AddSequencedLeavesRequest {
    int64 log_id = 1;
    repeated SequencedLogLeaf leaves = 2;
}
message SequencedLogLeaf {
    int64 leaf_index = 1;
    LogLeaf leaf = 2;
}
```
Both look a bit ugly to me. WDYT?

Reply:
I don't think it's a good idea to have fields that are ignored and it looks like the first option could be confusing. So the second alternative might be better?

I don't think we're required to export all the API methods to HTTP? Maybe we can just comment the new API as provisional for the moment and come back to it later. We should maybe make clear that the mirroring stuff is being developed and not ready for use yet.

What do you think?

Reply:
Both alternatives above are for the case when we ignore the `LogLeaf.LeafIndex` field. So if you think ignoring is a bad idea then the current code is the solution, and neither of the alternatives.

Good idea to add a comment, will do.

Reply:
I think ignoring is bad if that will cause confusion for clients. If we think the API is clear then maybe it's OK.